test/lib/jdk/test/lib/process/StreamPumper.java

```diff
@@ -133,6 +134,8 @@ public void run() {
 
                         i++;
                     }
+                    // The remaining after last '\n' (lastcrlf position) buffer data is written into lineBos.
+                    // The end of this line from next buffer is concatenated to this data in the next iteration.
```

Comment:
In trying to help my own understanding here I suggest:
```
// If no crlf was found, or there was additional data after the last crlf was found, then write the leftover data
// in lineBos. If there is more data to read it will be concatenated with the current data on the next iteration.
// If there is no more data, or no more crlf found, all the remaining data will be processed after the loop, as the
// final line.
```